Kitodo/src/main/java/org/kitodo/services/validation/MetadataValidationService.java

```diff
@@ -101,24 +101,29 @@ public boolean validate(Fileformat gdzfile, Prefs prefs, Process process) {
         }
 
         DocStruct logical = dd.getLogicalDocStruct();
-        if (logical.getAllIdentifierMetadata() != null && logical.getAllIdentifierMetadata().size() > 0) {
-            Metadata identifierTopStruct = logical.getAllIdentifierMetadata().get(0);
+        List<Metadata> allIdentifierMetadata = logical.getAllIdentifierMetadata();
+        if (allIdentifierMetadata != null && allIdentifierMetadata.size() > 0) {
+            Metadata identifierTopStruct = allIdentifierMetadata.get(0);
             try {
                 result = checkIfMetadataValueNotReplaced(logical, identifierTopStruct, metadataLanguage);
 
                 DocStruct firstChild = logical.getAllChildren().get(0);
-                Metadata identifierFirstChild = firstChild.getAllIdentifierMetadata().get(0);
-                if (identifierTopStruct.getValue() != null && !identifierTopStruct.getValue().isEmpty()
-                        && identifierTopStruct.getValue().equals(identifierFirstChild.getValue())) {
-                    List<String> parameter = new ArrayList<>();
-                    parameter.add(identifierTopStruct.getType().getName());
-                    parameter.add(logical.getType().getName());
-                    parameter.add(firstChild.getType().getName());
-                    Helper.setFehlerMeldung(Helper.getTranslation("InvalidIdentifierSame", parameter));
-                    result = false;
-                }
+                List<Metadata> allChildIdentifierMetadata = firstChild.getAllIdentifierMetadata();
+                if (allChildIdentifierMetadata != null && allChildIdentifierMetadata.size() > 0) {
+                    Metadata identifierFirstChild = allChildIdentifierMetadata.get(0);
+                    if (identifierTopStruct.getValue() != null && !identifierTopStruct.getValue().isEmpty()
+                            && identifierTopStruct.getValue().equals(identifierFirstChild.getValue())) {
+                        List<String> parameter = new ArrayList<>();
+                        parameter.add(identifierTopStruct.getType().getName());
+                        parameter.add(logical.getType().getName());
+                        parameter.add(firstChild.getType().getName());
+                        Helper.setFehlerMeldung(Helper.getTranslation("InvalidIdentifierSame", parameter));
+                        result = false;
+                    }
 
-                result = checkIfMetadataValueNotReplaced(firstChild, identifierFirstChild, metadataLanguage);
+                    result = checkIfMetadataValueNotReplaced(firstChild, identifierFirstChild, metadataLanguage);
+                }
+                //TODO: why catch here, if this block doesn't throw any exception??
```

Comment:
If this try catch block is not needed, then remove it.

Reply:
I have looked at this again and it looks this block was only for catching NPE and logging. SO I remove it and move logger to else block.